nemo/collections/nlp/modules/common/text_generation_server.py

```diff
@@ -142,11 +144,36 @@ def put(self):
             if min_tokens_to_generate < 0:
                 return "min_tokens_to_generate must be an integer no less than 0"
 
+        neighbors = None
+        if "neighbors" in request.get_json():
+            neighbors = request.get_json()["neighbors"]
+            if not isinstance(neighbors, int):
+                return "num of neighbors must be an integer no less than 0"
+            if neighbors < 0:
```

Comment:
is `neighbors=0` an acceptable number too? Would you not just ignore then?

Reply:
yes. neighbor = 0 is acceptable. In the case of K=0, we will use padding as context token, and the attention mask will mask them out.